nowriter/tmencoding/tm_encoder_easy_intr.go

```diff
@@ -0,0 +1,22 @@
+package tmencoding
+
+import "time"
+
+type TMEncoderEasyIntr interface {
```

Comment:
It is indeed a weaker form of BuilderIntr in many ways. The most important advantage it has is that it makes fewer assumptions about the methods. In that they are procedures (no return value) as opposed to builder. In many cases Builder completely subsumes this pattern but there are some cases where it doesn't. It's important to include here as an option because it's in some significant proportion of people's view considered simpler than Builder because some people are not as comfortable with the elegant chaining construction.

Reply:
there exists an east converter between it and several other of these interfaces. all those converters are boring and interchangeable which is the main power of the nearly redundant interfaces.

Reply:
i am going to do all the godoc now thanks i was wondering about it @ethanfrey